runners/spark/src/test/java/org/apache/beam/runners/spark/translation/StorageLevelTest.java

```diff
@@ -17,29 +17,29 @@
  */
 package org.apache.beam.runners.spark.translation;
 
-import org.apache.beam.runners.spark.PipelineRule;
+import org.apache.beam.runners.spark.SparkPipelineOptions;
+import org.apache.beam.runners.spark.SparkRunner;
 import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
 import org.apache.beam.sdk.testing.PAssert;
 import org.apache.beam.sdk.transforms.Count;
 import org.apache.beam.sdk.transforms.Create;
 import org.apache.beam.sdk.values.PCollection;
-import org.junit.Rule;
 import org.junit.Test;
 
 /**
  * Test the RDD storage level defined by user.
  */
 public class StorageLevelTest {
```

Comment:
OK, figured out, we had a bug causing this, it's fixed on master now so please rebase.
Then use your original implementation:
```java
public class StorageLevelTest {

  @Rule
  public final transient PipelineRule pipelineRule = PipelineRule.batch();

  @Test
  public void test() throws Exception {
    pipelineRule.getOptions().setStorageLevel("DISK_ONLY");
    Pipeline pipeline = pipelineRule.createPipeline();

    PCollection<String> pCollection = pipeline.apply(Create.of("foo"));

    // by default, the Spark runner doesn't cache the RDD if it accessed only one time.
    // So, to "force" the caching of the RDD, we have to call the RDD at least two time.
    // That's why we are using Count fn on the PCollection.
    pCollection.apply(Count.<String>globally());

    PCollection<String> output = pCollection.apply(new StorageLevelPTransform());

    PAssert.thatSingleton(output).isEqualTo("Disk Serialized 1x Replicated");

    pipeline.run();
  }

}```